src/backend/distributed/planner/extended_op_node_utils.c

```diff
@@ -62,22 +62,17 @@ BuildExtendedOpNodeProperties(MultiExtendedOp *extendedOpNode, bool
 		ShouldPullDistinctColumn(repartitionSubquery, groupedByDisjointPartitionColumn,
 								 hasNonPartitionColumnDistinctAgg);
 
-	/*
-	 * TODO: Only window functions that can be pushed down reach here, thus,
-	 * using hasWindowFuncs is safe for now. However, this should be fixed
-	 * when we support pull-to-master window functions.
-	 */
-	bool pushDownWindowFunctions = extendedOpNode->hasWindowFuncs;
-
 	extendedOpNodeProperties.groupedByDisjointPartitionColumn =
 		groupedByDisjointPartitionColumn;
 	extendedOpNodeProperties.repartitionSubquery = repartitionSubquery;
 	extendedOpNodeProperties.hasNonPartitionColumnDistinctAgg =
 		hasNonPartitionColumnDistinctAgg;
 	extendedOpNodeProperties.pullDistinctColumns = pullDistinctColumns;
-	extendedOpNodeProperties.pushDownWindowFunctions = pushDownWindowFunctions;
 	extendedOpNodeProperties.pullUpIntermediateRows =
 		!groupedByDisjointPartitionColumn && pullUpIntermediateRows;
+	extendedOpNodeProperties.hasWindowFuncs = extendedOpNode->hasWindowFuncs;
+	extendedOpNodeProperties.pushDownWindowFunctions =
```

Comment:
Maybe better to keep consistent names, keep this as `hasNonPushableWindowFunction`